frontend/src/Frontend/Setup/Browser.hs

```diff
@@ -0,0 +1,289 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecursiveDo #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Wallet setup screens
+module Frontend.Setup.Browser (runSetup, bipWalletBrowser) where
+
+import Control.Lens ((<>~), (^.), _1, _2, _3)
+import Control.Monad (guard)
+import Control.Monad.Fix (MonadFix)
+import Control.Monad.Trans (lift)
+import Control.Monad
+import Data.Aeson (ToJSON(..), FromJSON(..))
+import Data.Aeson.GADT.TH
+import Data.ByteString (ByteString)
+import Data.Constraint.Extras.TH
+import Data.Dependent.Sum
+import Data.Functor.Compose
+import Data.Functor.Identity
+import Data.GADT.Compare.TH
+import Data.GADT.Show.TH
+import Data.Traversable (for)
+import Data.Universe.Some.TH
+import Language.Javascript.JSaddle (MonadJSM)
+import Reflex.Dom.Core hiding (Key)
+import Pact.Server.ApiClient (HasTransactionLogger, askTransactionLogger, _transactionLogger_rotateLogFile)
+import Obelisk.Route.Frontend
+import Obelisk.Generated.Static
+import Common.Wallet
+import Common.Route
+import qualified Frontend.App as App (app)
+import Frontend.AppCfg
+import Frontend.Crypto.Class
+import Frontend.Crypto.Password
+import Frontend.Crypto.Ed25519
+import Frontend.Crypto.Browser
+import Frontend.Foundation
+import Frontend.Setup.Common
+import Frontend.Setup.Password
+import Frontend.Setup.Widgets
+import Frontend.Storage
+import Frontend.UI.Button
+import Frontend.UI.Modal.Impl (showModalBrutal)
+import Frontend.UI.Dialogs.LogoutConfirmation (uiIdeLogoutConfirmation)
+import Frontend.UI.Widgets
+import Frontend.VersionedStore
+
+
+data BIPStorage a where
```

Comment:
Does this really need to be a GADT?  If this was duplicated from somewhere else, then not a blocker.

Reply:
Copied from `desktop/src/Desktop/Crypto/BIP.hs`